xfrd-tcp.c

```diff
@@ -775,6 +992,104 @@ xfrd_tcp_write(struct xfrd_tcp_pipeline* tp, xfrd_zone_type* zone)
 	tcp_pipe_reset_timeout(tp);
 }
 
+#ifdef HAVE_TLS_1_3
+static int
+conn_read_ssl(struct xfrd_tcp* tcp, SSL* ssl)
+{
+	ssize_t received;
+	/* receive leading packet length bytes */
+	if(tcp->total_bytes < sizeof(tcp->msglen)) {
+		ERR_clear_error();
+		received = SSL_read(ssl,
+						(char*) &tcp->msglen + tcp->total_bytes,
+						sizeof(tcp->msglen) - tcp->total_bytes);
+		if (received <= 0) {
+			int err = SSL_get_error(ssl, received);
+			log_msg(LOG_ERR, "ssl_read returned error %d", err);
```

Comment:
I would put the log_msg after the err zero check below, no need to log an error when the connection just closes.  Perhaps on high verbosity or debug, but not for regular operations.

Reply:
Resolved in current code.